### PR TITLE
Several, small documentation improvements

### PR DESCRIPTION
**Pull request by mmattar**
- Re-organize main repo README
- Added links in ML-Agents-Overview page to other parts of the docs that show the "how to"
- Minor clean-ups to Python package-specific readme files
- Clean-up to Unity Inference Engine page
- Update to the docs README
- Added a specific cross-platform section in ML-Agents Overview to amplify Barracuda
- Updated the links in Limitations.md to point to the specific subsections
- Cleaned up the Designing a Learning Environment page. Added an intro paragraph.
- Updated the installation guide to specifically call out local installation
- A few minor formatting, spelling errors fixed.

### Proposed change(s)

Describe the changes made in this PR.

### Types of change(s)

- [ ] Bug fix
- [ ] New feature
- [ ] Code refactor
- [ ] Breaking change
- [X] Documentation update
- [ ] Other (please describe)

### Checklist
- [ ] Added tests that prove my fix is effective or that my feature works
- [ ] Updated the [changelog](https://github.com/Unity-Technologies/ml-agents/blob/master/com.unity.ml-agents/CHANGELOG.md) (if applicable)
- [X] Updated the [documentation](https://github.com/Unity-Technologies/ml-agents/tree/master/docs) (if applicable)
- [ ] Updated the [migration guide](https://github.com/Unity-Technologies/ml-agents/blob/master/docs/Migrating.md) (if applicable)

### CODE REVIEW

File: docs/ML-Agents-Overview.md [hunk 1]
```diff
@@ -594,18 +612,17 @@ learns, the ML-Agents Toolkit provides a way to randomly sample parameters of
 the environment during training. We refer to this approach as **Environment
 Parameter Randomization**. For those familiar with Reinforcement Learning
 research, this approach is based on the concept of Domain Randomization (you can
-read more about it [here](https://arxiv.org/abs/1703.06907)). By using parameter
-randomization during training, the agent can be better suited to adapt (with
+read more about it [here](https://arxiv.org/abs/1703.06907)). By using
```
Comment (chriselion) [inline docs/ML-Agents-Overview.md:615]: nit: just link "Domain Randomization" to the arxiv paper? (and remove "you can read more about it here")

File: docs/Unity-Inference-Engine.md [hunk 2]
```diff
@@ -1,8 +1,9 @@
 # Unity Inference Engine
 
 The ML-Agents Toolkit allows you to use pre-trained neural network models
-inside your Unity games. This support is possible thanks to the Unity Inference
-Engine. The Unity Inference Engine is using
+inside your Unity games. This support is possible thanks to the
+[Unity Inference Engine](https://docs.unity3d.com/Packages/com.unity.barracuda@0.7/manual/index.html)
```
Comment (chriselion) [inline docs/Unity-Inference-Engine.md:5]:
```suggestion
[Unity Inference Engine](https://docs.unity3d.com/Packages/com.unity.barracuda@latest/index.html)
```
Just link to the latest version

File: docs/Unity-Inference-Engine.md [hunk 3]
```diff
@@ -11,21 +12,16 @@ trainers.
 
 ## Supported devices
 
+See the Unity Inference Engine documentation for a list of the
+[supported platforms](https://docs.unity3d.com/Packages/com.unity.barracuda@0.7/manual/index.html#supported-platforms).
```
Comment (chriselion) [inline docs/Unity-Inference-Engine.md:16]: same

File: ml-agents-envs/README.md [hunk 4]
```diff
@@ -2,32 +2,35 @@
...
 ## Limitations
- - `mlagents_envs` uses localhost ports to exchange data between Unity and Python. As such,
- multiple instances can have their ports collide leading to errors. Make sure to use a
- different port if you are using multiple instances of `UnityEnvironment`.
- - Communication between Unity and the Python `UnityEnvironment` is not secure.
- - On Linux, ports are not released immediately after the communication closes. As such, you
- cannot reuse ports right after closing a `UnityEnvironment`.
+
+- `mlagents_envs` uses localhost ports to exchange data between Unity and
+  Python. As such, multiple instances can have their ports collide leading to
```
Comment (chriselion) [inline ml-agents-envs/README.md:31]:
```suggestion
  Python. As such, multiple instances can have their ports collide, leading to
```

File: docs/ML-Agents-Overview.md (hunk 1 above)
Comment (mmattar) [reply docs/ML-Agents-Overview.md:615]: Done.

File: docs/Unity-Inference-Engine.md (hunk 2 above)
Comment (mmattar) [reply docs/Unity-Inference-Engine.md:5]: Good catch.

**Pull request merged**